pkg/executor/chained_task_handle_test.go

```diff
@@ -0,0 +1,193 @@
+package executor
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const taskExecutionTime = 100 * time.Millisecond
+const waitTimeout = 60 * time.Second
+
+func TestSuccsessfulChainedTaskHandle(t *testing.T) {
+	Convey("When having ChainedTaskHandle with successful execution", t, func() {
+		initialTaskHandle := new(MockTaskHandle)
+		initialTaskHandle.On("Wait", 0*time.Second).After(taskExecutionTime).Return(true, nil)
+		initialTaskHandle.On("Stop").Return(nil)
+
+		chainedTaskHandle := new(MockTaskHandle)
+		chainedTaskHandle.On("Wait", 0*time.Second).After(2*taskExecutionTime).Return(true, nil)
+		chainedTaskHandle.On("Stop").Return(nil)
+
+		chainedLauncher := new(MockLauncher)
+		chainedLauncher.On("Launch").Return(chainedTaskHandle, nil)
+
+		Convey("Successful run should yield no error", func() {
+			startTime := time.Now()
+			taskHandle := NewChainedTaskHandle(initialTaskHandle, chainedLauncher)
+			So(taskHandle.Status(), ShouldEqual, RUNNING)
+
+			isTerminated, err := taskHandle.Wait(waitTimeout)
+			So(time.Since(startTime), ShouldBeGreaterThanOrEqualTo, taskExecutionTime)
```

Comment:
@skonefal shouldn't it be `ShouldBeGreaterThanOrEqualTo, 3*taskExecutionTime` (sum of duration of all `Wait()` calls)? I think that `TestFailureChainedTaskHandle()` can prove that all the tasks were waited for; this test looks flaky to me (might fail when expected).